### PR TITLE
Export notification state to file for use with systray

### PR DESCRIPTION
**Pull request by cfcs**
Generates some warning messages because of the Lwt.bind-freakshow going on in the file writing function.
I've still no idea how Lwt is supposed to be used, but this works for my purposes. Rewrites welcome :-)

### CODE REVIEW

File: cli/cli_state.ml [hunk 1]
```diff
@@ -28,7 +42,56 @@ type ui_state = {
...
+  (* this function should be private; not included in the mli *)
+  let lwt_crap () =
+  let s  = string_of_session_status session_state in
+  lwt _ =
+  Lwt.bind
+    (Lwt.bind
+      (Lwt.bind
+        (Lwt.bind
+          (Lwt.bind
+            (Lwt_unix.openfile
```
Comment (samoht) [inline cli/cli_state.ml]: This is quite ugly :p 

Usually, people use the `>>=` operator:
```
Lwt_unix.openfile ... >>= fun fd ->
let oc = Lwt_io.of_fd Output fd in
Lwt_io.set_positition ... >>= fun () ->
Lwt_io.write_line ... >>= fun () ->
...
```

**Comment by hannesm** [comment]
while I appreciate this contribution, I'm a bit uneasy with its current implementation (using Lwt.join and spreading all over). Instead, I'd do the following (and will if time permits):
- use a React.event (or signal) for notifications
- have a listener via a channel that waits on this (or a second signal) and when something arrives, writes to a file (inherently not racy since there is only one listener!)

This would cleanup the mutable notifications, and move the file writing into a single place.

**Comment by hannesm** [comment]
closing now that #55 (or #56) hit master. thanks!

**Comment by cfcs** [comment]
Awesome!

**Pull request closed without merging**